### DISCUSSION TITLE
How to parse Simple Chinese PDF

### DISCUSSION THREAD

**Discussion by wanghang-github** [pymupdf/PyMuPDF#2536] [Q&A]
How to parse Simple Chinese PDF

**Comment by little-polka-dot** [comment]
简体中文pdf解析有什么问题吗😊

**Comment by wanghang-github** [comment]
是这样，我想解析一些中文论文，文章存在一些页眉页脚和图片部分的内容
![image](https://github.com/pymupdf/PyMuPDF/assets/69884447/c91dbbc5-d567-48b3-bc9f-68025ddfc654)

在解析出来之后出现一些比较乱的英文字母，
![image](https://github.com/pymupdf/PyMuPDF/assets/69884447/205e90a7-bad3-4b96-bb2d-d07810450d01)

我又应该如何根据字体识别出来各个章节呢。
![image](https://github.com/pymupdf/PyMuPDF/assets/69884447/1251f13b-70cb-4c44-83ca-8c7c6c80560e)
哈哈哈，同胞啊

**Comment by little-polka-dot** [comment]
识别各个章节要么通过坐标计算的方式，要么通过视觉模型识别区域，如果你是想过滤掉页眉页脚的话，只能先对整页进行视觉版面识别。pdf本身并没有这些信息。

**Comment by little-polka-dot** [comment]
我的工作刚好是做这个的

**Comment by little-polka-dot** [comment]
至于图片下的乱码，可以获取图片坐标与seqno，获取文本的坐标和seqno，然后判断文本是不是被图片覆盖了，如果覆盖了则是不可见文本。

**Comment by little-polka-dot** [comment]
get_texttrace获取文本，get_bboxlog获取图片坐标

**Comment by little-polka-dot** [comment]
计算是否是不可见文本有以下规则
Invisibility may happen for a number of reasons:
        1) text covered by something else (image, vector graphics)
        2) text has same color as background (like black on black)
        3) text is under control of rendering mode 3 = 3 Tr.
           This is very often used by OCR: recognized text is written to the page with this rendering mode -
           at the same spot where the text in the image was found, thus making it selectable / searchable.
但是目前我发现了一个疑似bug，就是get_texttrace返回的坐标不对，作者还没有回复

**Comment by JorjMcKie** [comment]
I am not sure what to make out of this post.
After using a Chinese-English translator, it appears however, that we have a typical "Discussions" thread here.
May I also ask for sticking with English as the *lingua franca* for IT?

**Reply by little-polka-dot** [reply]
😀😀As we all know, the vast majority of people in mainland China are not very good at English.

**Reply by wanghang-github** [reply]
Yes,I'm not good at English.Can I get yout wechat,you could add me Wh19956486056

**Reply by little-polka-dot** [reply]
Me? 17721441649
